### PR TITLE
[IMM32] Rewrite ImmUnregisterWordW

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Implementing Japanese input...
JIRA issue: [CORE-11700](https://jira.reactos.org/browse/CORE-11700)

## Proposed changes

- Rewrite `ImmUnregisterWordW` function.

### CODE REVIEW

File: dll/win32/imm32/imm.c [hunk 1]
```diff
@@ -4107,28 +4107,56 @@ BOOL WINAPI ImmUnregisterWordA(
...
+        cchW = lstrlenW(lpszUnregister);
+        cchA = (cchW + 1) * 2;
+        pszUnregisterA = Imm32HeapAlloc(0, cchA);
+        if (!pszUnregisterA)
+            goto Quit;
+        cchA = WideCharToMultiByte(CP_ACP, 0, lpszUnregister, cchW, pszUnregisterA, cchA, NULL, NULL);
+        pszUnregisterA[cchA] = 0;
+    }
+
+    ret = ImmUnregisterWordA(hKL, pszReadingA, dwStyle, pszUnregisterA);
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:4151]: Usually it's more natural to implement the wide function, then to implement the ANSI function by converting the strings to wide unicode and calling the wide function; not the reverse.

File: dll/win32/imm32/imm.c [hunk 2]
```diff
@@ -4107,28 +4107,56 @@ BOOL WINAPI ImmUnregisterWordA(
 BOOL WINAPI ImmUnregisterWordW(
   HKL hKL, LPCWSTR lpszReading, DWORD dwStyle, LPCWSTR lpszUnregister)
 {
-    ImmHkl *immHkl = IMM_GetImmHkl(hKL);
-    TRACE("(%p, %s, %d, %s):\n", hKL, debugstr_w(lpszReading), dwStyle,
-            debugstr_w(lpszUnregister));
-    if (immHkl->hIME && immHkl->pImeUnregisterWord)
+    PIMEDPI pImeDpi;
+    LPSTR pszReadingA = NULL, pszUnregisterA = NULL;
+    BOOL ret = FALSE;
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:4112]:
```suggestion
    BOOL ret = FALSE;
    PIMEDPI pImeDpi;
    LPSTR pszReadingA = NULL, pszUnregisterA = NULL;
```

Comment (katahiromz) [reply dll/win32/imm32/imm.c]: OK.

File: dll/win32/imm32/imm.c (hunk 1 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c:4151]: But this is the case of `!(fdwProperty & IME_PROP_UNICODE)` (non-Unicode property).

Comment (katahiromz) [reply dll/win32/imm32/imm.c]: Improved.

**Pull request merged**